src/main/java/de/tum/in/www1/artemis/service/compass/CompassService.java

```diff
@@ -186,23 +184,28 @@ public Grade getResultForModel(long exerciseId, long studentId, long modelId) {
     /**
      * Add an assessment to an engine
      *
-     * @param exerciseId the exerciseId
-     * @param modelId the corresponding modelId
-     * @param assessment the new assessment as raw string
+     * @param exerciseId         the exerciseId
+     * @param submissionId       the corresponding modelId
+     * @param modelingAssessment the new assessment as raw string
      */
-    public void addAssessment(long exerciseId, long modelId, String assessment) {
-        log.info("Add assessment for exercise" + exerciseId + " and model " + modelId);
-        if (!loadExerciseIfSuspended(exerciseId)) {
+    public void addAssessment(long exerciseId, long submissionId, List<ModelElementAssessment> modelingAssessment) {
+        log.info("Add assessment for exercise" + exerciseId + " and model " + submissionId);
+        if (!loadExerciseIfSuspended(exerciseId)) { //TODO rewordk after distinguishing between saved and submitted assessments on filesystem
             return;
         }
         CalculationEngine engine = compassCalculationEngines.get(exerciseId);
-        engine.notifyNewAssessment(assessment, modelId);
+        engine.notifyNewAssessment(modelingAssessment, submissionId);
         // Check all models for new assessments
-        for (long id: engine.getModelIds()) {
+        for (long id : engine.getModelIds()) {
             assessAutomatically(id, exerciseId);
         }
     }
 
+    public List<Conflict> checkForConflicts(long exerciseId, long submissionId, List<ModelElementAssessment> modelingAssessment) {
```

Comment:
This method is called checkForConflitcts and the one in the engine is called getConflicts. I would suggest to call them both getConflicts. In the case there are no conflicts you simply return an empty list (or null)

Reply:
Yep you are right. Renamed it :)